app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -421,6 +424,155 @@ private void loadIcal(BufferedReader in) {
 		}
 	}
 
+	/*Reading the JSON open event format data fetched from the API end point*/
+	private void loadJson(BufferedReader in) {
+
+		StringBuffer buffer = new StringBuffer();
+		SimpleDateFormat df = new SimpleDateFormat();
+		HashMap<String, Schedule.Line> tentMap = new HashMap<String, Schedule.Line>();
+		Scanner s = new Scanner(in);
+
+		while (s.hasNext()) {
+			buffer.append(s.nextLine());
+
+		}
+
+		String output = buffer.toString();
+		Schedule.Line line = null;
+
+		try {
+
+			JSONObject conference = new JSONObject(output);
+			int i;
+
+			//Using this title name so the user doesn't see URL when she clicks back
+			title = conference.getString("name");
+
+			if (conference.has("logo")) {
+				icon = conference.getString("logo");
+			}
+
+			if (conference.has("social_links")) {
+				links = new LinkedList<>();
+				JSONArray linklist = conference.getJSONArray("social_links");
+				for (i = 0; i < linklist.length(); ++i) {
+					JSONObject link = linklist.getJSONObject(i);
+					Schedule.Link slink = new Link(link.getString("link"), link.getString("name"));
+					slink.setType(link.optString("type", null));
+					links.addLast(slink);
+				}
+
+				//Adding website separately because it is not in social_links
+				if (conference.has("event_url")) {
+					Schedule.Link slink = new Link(conference.getString("event_url"), "Website");
```

Comment:
I also think that should go that way, was a little sceptical about it. No problem :+1: